go/test/endtoend/prepare-statement/main_test.go

```diff
@@ -0,0 +1,272 @@
+		t_timestamp TIMESTAMP,
+		c8 bit(8) DEFAULT NULL,
+		c16 bit(16) DEFAULT NULL,
+		c24 bit(24) DEFAULT NULL,
+		c32 bit(32) DEFAULT NULL,
+		c40 bit(40) DEFAULT NULL,
+		c48 bit(48) DEFAULT NULL,
+		c56 bit(56) DEFAULT NULL,
+		c63 bit(63) DEFAULT NULL,
+		c64 bit(64) DEFAULT NULL,
+		json_col JSON,
+		text_col TEXT,
+		data longblob,
+		primary key (id)
+		) Engine=InnoDB`
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+
+	exitcode, err := func() (int, error) {
+		clusterInstance = cluster.NewCluster(cell, hostname)
+
+		defer clusterInstance.Teardown()
+
+		// Start topo server
+		if err := clusterInstance.StartTopo(); err != nil {
+			return 1, err
+		}
+
+		// create auth server config
+		SQLConfig := `{
+			"testuser1": {
+				"Password": "testpassword1",
+				"UserData": "vtgate client 1"
+			}
+		}`
+		if err := createConfig(mysqlAuthServerStatic, SQLConfig); err != nil {
+			return 1, err
+		}
+
+		// add extra arguments
+		clusterInstance.VtGateExtraArgs = []string{
+			"-mysql_auth_server_impl", "static",
+			"-mysql_server_query_timeout", "1s",
+			"-mysql_auth_server_static_file", clusterInstance.TmpDirectory + mysqlAuthServerStatic,
+			"-mysql_server_version", "8.0.16-7",
+		}
+
+		// Start keyspace
+		keyspace := &cluster.Keyspace{
+			Name:      keyspaceName,
+			SchemaSQL: sqlSchema,
+		}
+		if err := clusterInstance.StartUnshardedKeyspace(*keyspace, 1, false); err != nil {
+			return 1, err
+		}
+
+		// Start vtgate
+		if err := clusterInstance.StartVtgate(); err != nil {
+			return 1, err
+		}
+
+		dbInfo.Host = clusterInstance.Hostname
+		dbInfo.Port = uint(clusterInstance.VtgateMySQLPort)
+
+		return m.Run(), nil
+	}()
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		os.Exit(1)
+	} else {
+		os.Exit(exitcode)
+	}
+
+}
+
+// ConnectionString generates the connection string using dbinfo.
+func (db DBInfo) ConnectionString(params ...string) string {
+	return fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?%s", db.Username, db.Password, db.Host,
+		db.Port, db.KeyspaceName, strings.Join(append(db.Params, params...), "&"))
+}
+
+// createConfig create file in to Tmp dir in vtdataroot and write the given data.
+func createConfig(name, data string) error {
+	// creating new file
+	f, err := os.Create(clusterInstance.TmpDirectory + name)
+	if err != nil {
+		return err
+	}
+
+	if data == "" {
+		return nil
+	}
+
+	// write the given data
+	_, err = fmt.Fprint(f, data)
+	return err
+}
+
+// GetORM will connect the vtgate through mysql protocol.
+func GetORM(t *testing.T, params ...string) *sql.DB {
+	dbo, err := sql.Open("mysql", dbInfo.ConnectionString(params...))
+	require.Nil(t, err)
+	return dbo
+}
+
+// execWithIgnore executes the prepared query, and ignore the given error codes.
+func execWithIgnore(t *testing.T, dbo *sql.DB, errorCodes []uint16, stmt string, params ...interface{}) {
+	if _, err := dbo.Exec(stmt, params...); err != nil {
+		require.Contains(t, errorCodes, err.(*mysql.MySQLError).Number)
+		glog.Info("error_ignored", err)
```

Comment:
Are we adding a dependency to `glog` or is this an existing dependency?
In any case, we should be using only `t.Logf` methods, not anything else.